lib/perl/Genome/Site/TGI/Extension/UnarchiveInstrumentData.pm

```diff
@@ -0,0 +1,196 @@
+        unless (
+            Genome::Sys->rsync_directory(
+                source_directory => $unarchived_bam_dirname,
+                target_directory => $allocation_bam_dirname,
+            )
+          ) {
+            $self->fatal_message('Failed to rsync from %s to %s', $unarchived_bam_dirname, $allocation_bam_dirname);
+        }
+        unless ($ENV{UR_DBI_NO_COMMIT}) {
+            Genome::Sys::CommitAction->create(
+                on_commit => sub {
+                    unless(Genome::Sys->remove_directory_tree($unarchived_bam_dirname)) {
+                        $self->fatal_message('Failed to remove directory: %s', $unarchived_bam_dirname);
+                    }
+                }
+            );
+        }
+        $allocation->reallocate();
+        UR::Context->commit();
+
+        my $allocated_bam_path = File::Spec->join($allocation_bam_dirname,$unarchived_bam_filename);
+        if (-e $allocated_bam_path) {
+            $instrument_data->bam_path($allocated_bam_path);
+            UR::Context->commit();
+        } else {
+            $self->fatal_message('Failed to find the new allocated BAM path: %s', $allocated_bam_path);
+        }
+    }
+}
+
+sub resolve_lims_unarchive_script_path {
+    my $self = shift;
+
+    my $abs_path = File::Spec->rel2abs(__FILE__);
+    my ($filename,$dirname) = File::Basename::fileparse($abs_path);
+    my $lims_unarchive_script_path = File::Spec->join($dirname,'generate_lims_disk_unarchive.pl');
+    unless (-e $lims_unarchive_script_path) {
+        $self->fatal_message('Failed to resolve the path to the LIMS script required for unarchiving.');
+    }
+    return $lims_unarchive_script_path;
+}
+
+sub get_or_create_allocation {
+    my $self = shift;
+    my $instrument_data = shift;
+    my %params = (
+        disk_group_name => 'info_alignments',
+        allocation_path => File::Spec->join('instrument_data',$instrument_data->id),
+        kilobytes_requested => 12642864,
+        owner_class_name => $instrument_data->class,
+        owner_id => $instrument_data->id,
+    );
+    my ($allocation) = Genome::Disk::Allocation->get(allocation_path => $params{allocation_path});
+    if ($allocation) {
+        return $allocation;
+    } else {
+        my $create_cmd = Genome::Disk::Command::Allocation::Create->create(%params);
+        unless ($create_cmd->execute) {
+            $self->fatal_message('Could not create allocation for instrument data: %s', $instrument_data->id);
+        }
+        ($allocation) = Genome::Disk::Allocation->get(allocation_path => $params{allocation_path});
+        unless ($allocation) {
+            $self->fatal_message('Could not find allocation for instrument data: %s', $instrument_data->id);
+        }
+    }
+    return $allocation;
+}
+
+sub validate_bam_is_complete {
+    my $self = shift;
+    my $bam_path = shift;
+    my $instrument_data = shift;
+
+    if (-e $bam_path .'.md5') {
+        print STDERR 'Calculating md5 for BAM: '. $bam_path ."\n";
```

Comment:
Only that I missed this one one when converting the script to a real module... Changed!